Misc/NEWS.d/next/Library/2023-01-06-21-14-41.gh-issue-100809.I697UT.rst

```diff
@@ -0,0 +1,3 @@
+Fix handling of drive-relative paths (like 'C:' and 'C:foo') in
+:meth:`pathlib.Path.absolute`. This method now calls
+``nt._getfullpathname()`` on Windows.
```

Comment:
```suggestion
:meth:`pathlib.Path.absolute`. This method now uses the OS API
to retrieve the correct current working directory for the drive.
```

No need to refer to the internal API. We _could_ refer to the OS API (`GetFullPathNameW`), but I don't think that adds any real value.